cumulus_library/studies/core/core_templates/medicationrequest.sql.jinja

```diff
@@ -1,89 +1,123 @@
+        FROM medicationrequest AS mr
+    ),
+
+    contained_refs AS (
+        SELECT DISTINCT
+            mr.id,
+            substring(mr.med_ref, 2) AS medication_id
+        FROM mr_basics AS mr
+        WHERE mr.med_ref IS NOT NULL AND {{ syntax.like('mr.med_ref', '#%') }}
+    ),
 
-temp_mr AS (
-    SELECT 
-        {{- utils.basic_cols(
-                'medicationrequest',
-                'mr',
-                [
-                    'id', 
-                    'status', 
-                    'intent',
-                ]
-            ) 
-        }},
-        {{- utils.date_cols_from_str(
-                'medicationrequest',
-                'mr',
-                ['authoredOn'],
-                schema
-            ) 
-        }},
-        {{- utils.truncate_date_cols(
-                'medicationrequest',
-                'mr',
-                [
-                    ('authoredOn', 'month'),
-                ],
-                schema
-            ) 
-        }},
-        {{- utils.nullable_cols(
-                'medicationrequest',
+    external_refs AS (
+        SELECT DISTINCT
+            mr.id,
+            substring(mr.med_ref, 12) AS medication_id
+        FROM mr_basics AS mr
+        WHERE mr.med_ref IS NOT NULL AND {{ syntax.like('mr.med_ref', 'Medication/%') }}
+    ),
 
-                'mr',
-                [
-                    'reportedBoolean',
-                    ('subject', 'reference', 'subject_ref'),
-                    ('encounter', 'reference', 'encounter_ref'),
-                ],
-                schema
-        ) -}},
-        mrc.code AS category_code,
-        mrc.code_system AS category_system,
-        mrm.code AS medication_code,
-        mrm.code_system AS medication_system,
-        mrm.display AS medication_display,
-        mrdt.dosageInstruction_text
-    FROM medicationrequest AS mr
-    LEFT JOIN core__medicationrequest_dn_category AS mrc ON mr.id = mrc.id
-    LEFT JOIN core__medicationrequest_dn_inline_code AS mrm ON mr.id = mrm.id
-    LEFT JOIN tmp_dn_dosage_text AS mrdt ON mr.id = mrdt.id
-    WHERE mrm.code_system = 'http://www.nlm.nih.gov/research/umls/rxnorm'
-)
+    {# this holds all the fields that each unioned table below will share -#}
+    mr_shared AS (
+        SELECT
+            mr.id,
+            mr.status,
+            mr.intent,
+            mrc.code AS category_code,
+            mrc.code_system AS category_code_system,
+            mrc.display AS category_display,
+            mr.reportedBoolean,
+            mr.reported_ref,
+            mr.subject_ref,
+            mr.encounter_ref,
+            mr.authoredOn,
+            mr.authoredOn_month
+        FROM mr_basics AS mr
+        LEFT JOIN core__medicationrequest_dn_category AS mrc ON mr.id = mrc.id
+    )
+
+    {# Internal: medication data from inline ETL extraction.
+    95% of the time, this dataset is the 'correct' dataset for Cerner.
+    It may not be present in EPIC datasets. #}
+    SELECT
+        mr.*,
```

Comment:
Did you run sqlfluff on this? generally star selects are flagged by it.

Reply:
seems like it passed ci... i'm honestly very surprised. is it because of the interstitial table you're creating is inside this query? if so that's very good to know.